src/NerdBank.GitVersioning/ManagedGit/GitPackReader.cs

```diff
@@ -44,7 +44,7 @@ public static Stream GetObject(GitPack pack, Stream stream, long offset, string
             if (type == GitPackObjectType.OBJ_OFS_DELTA)
             {
                 var baseObjectRelativeOffset = ReadVariableLengthInteger(stream);
-                var baseObjectOffset = (int)(offset - baseObjectRelativeOffset);
+                var baseObjectOffset = offset - baseObjectRelativeOffset;
```

Comment:
Example: Offset is `4 000 000 000`, relative offset is `300 000`. The result would be incorrectly truncated into wrong value.

Reply:
nit: the type is no longer obvious here. Since we had a bug in the type, it seems useful to make it explicit. Perhaps you can make L46-L47 both explicit. For example:

```suggestion
                long baseObjectOffset = offset - baseObjectRelativeOffset;
```